verification/policy.go

```diff
@@ -203,3 +203,44 @@ func ValidatePolicyDocument(policyDoc *PolicyDocument) error {
 	// No errors
 	return nil
 }
+
+// getApplicableTrustPolicy returns a pointer to the deep copied TrustPolicy statement that applies to the given
+// registry URI. If no applicable trust policy is found, returns an error
+// see https://github.com/notaryproject/notaryproject/blob/main/trust-store-trust-policy-specification.md#selecting-a-trust-policy-based-on-artifact-uri
+func (policyDoc *PolicyDocument) getApplicableTrustPolicy(artifactUri string) (*TrustPolicy, error) {
+
+	artifactPath, err := getArtifactPathFromUri(artifactUri)
+	if err != nil {
+		return nil, err
+	}
+
+	var wildcardPolicy *TrustPolicy
+	var applicablePolicy *TrustPolicy
+	for _, policyStatement := range policyDoc.TrustPolicies {
+		if isPresent(wildcard, policyStatement.RegistryScopes) {
+			wildcardPolicy = policyStatement.deepCopy() // we need to deep copy because we can't use the loop variable address. see https://stackoverflow.com/a/45967429
+		} else if isPresent(artifactPath, policyStatement.RegistryScopes) {
+			applicablePolicy = policyStatement.deepCopy()
+		}
```

Comment:
qq: Where are we validating that there is only one applicable trust policy fir a given repository?

Reply:
https://github.com/notaryproject/notation-go/blob/main/verification/policy.go#L69